Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -245,18 +245,18 @@ lemma variableChange_comp (C C' : VariableChange R) (W : WeierstrassCurve R) :
     W.variableChange (C.comp C') = (W.variableChange C').variableChange C := by
   simp only [VariableChange.comp, variableChange]
   ext <;> simp only [mul_inv, Units.val_mul]
-  · linear_combination (norm := ring1) C.u⁻¹ * C.s * 2 * C'.u.inv_mul
+  · linear_combination (norm := ring1) ↑C.u⁻¹ * C.s * 2 * C'.u.inv_mul
   · linear_combination (norm := ring1)
-      C.s * (-C'.s * 2 - W.a₁) * C.u⁻¹ ^ 2 * C'.u⁻¹ * C'.u.inv_mul
+      C.s * (-C'.s * 2 - W.a₁) * C.u⁻¹ ^ 2 * ↑C'.u⁻¹ * C'.u.inv_mul
         + (C.r * 3 - C.s ^ 2) * C.u⁻¹ ^ 2 * pow_mul_pow_eq_one 2 C'.u.inv_mul
   · linear_combination (norm := ring1)
-      C.r * (C'.s * 2 + W.a₁) * C.u⁻¹ ^ 3 * C'.u⁻¹ * pow_mul_pow_eq_one 2 C'.u.inv_mul
+      C.r * (C'.s * 2 + W.a₁) * C.u⁻¹ ^ 3 * ↑C'.u⁻¹ * pow_mul_pow_eq_one 2 C'.u.inv_mul
```

Comment:
Why did this end up changing?

Reply:
I think I looked at this earlier and from top of my head:

Previously the tactic would parse the `\^-1` before the term `C.u` was parsed (and translate it into using one of the theorems that got removed) but now it's actually the normal term parser parsing the entire `C.u\^-1`.

Therefore, now it thinks the inverse should be taken in `R` instead of in `R\^x` without the coersion hint (which is btw also displayed in the goal). This seems like normal Lean behaviour so the coersion hints are in my opinion expected there.

Reply:
Previously `⁻¹` was one of the supported operations *on hypotheses*, so `⁻¹` got handled differently *in constants* too: the atom which got pre-elaborated was `C'.u`.  Now the atom which gets pre-elaborated is `C'.u⁻¹`.  (Though I don't actually see why it makes a difference here!)

Reply:
Thanks, I've copied @hrmacbeth's version into the PR description.